src/ert/_c_wrappers/enkf/enkf_main.py

```diff
@@ -171,6 +174,22 @@ def runpaths(self):
     def runpath_list_filename(self):
         return self._runpaths.runpath_list_filename
 
+    @property
+    def storage(self):
+        return self._fs
+
+    @storage.setter
+    def storage(self, file_system):
+        self.addDataKW("<ERT-CASE>", file_system.getCaseName())
+        self.addDataKW("<ERTCASE>", file_system.getCaseName())
+        for key in file_system.getSummaryKeySet().keys():
+            self.ensembleConfig().add_summary(key)
```

Comment:
Also quite strange behavior, does this make sense? Some tests rely on it, not sure about real cases.

Reply:
This is indeed weird and unwanted. The configs should preferably hold no state so it would be better to move in that direction. Perhaps have a look at the references to ensemble_config_get_node and try to decifer why this is needed?

Reply:
Worth noting that all the integration tests worked fine without this, so could remove it and change the failing tests. However might be better to create an issue to fix this after this PR? My idea was to do this with minimal changing of existing tests, and then do some smaller follow up PRs. Does that sound reasonable?

Reply:
I agree, lets not change it.